### ISSUE TITLE
When there is no language dependent CU

### ISSUE THREAD

**Issue by andreasblueher** [brianlala/AutoSPSourceBuilder#31]
Hello Brian, 

I just created a new July 2018 source with slipstreamed updates for SharePoint 2016. Since July 2018 CU doesn't contain language dependent file, shouldn't it use the next one instead of not installing anything?

I hope you understand what I mean. Cheers, Andreas

**Comment by brianlala** [comment]
Hi Andreas, July 2018 hasn't even been added to the XML yet - or are you using your own updated copy?

**Comment by brianlala** [comment]
But I think I know what you mean... if we only include the language-independent package for July 2018, we'll have no language-dependent fixes at all. So the proper thing would be to look for the most recent language-dependent fix and install that, along with the requested language-independent fix... this could be challenging to implement! 

Also could put us in an interesting mixed-patch situation where we'd have the language-dependent fix from one month and the language-independent fix from another month (which, although it's probably supported, may not be what the user intended).

**Comment by andreasblueher** [comment]
I created an PR for the July updates, so yes I used my own copy.

Regarding the mixed-patch situation, isn't that also true with the current situation. It takes SP2016 RTM and uses July language independent update, but no language dependent updates at all. Shouldn't the language dependent files be updated to at least the latest current version, which would be the last month version?

Stefan Goßner states in his latest [blog](https://blogs.technet.microsoft.com/stefan_gossner/2018/07/10/july-2018-cu-for-sharepoint-server-2016-is-available-for-download/) for SP2016 

> check June 2018 CU for latest language dependent fixes

which to me indicates that you should indeed combine the June 2018 language dependent CU files with July 2018 language independent CU files

**Comment by brianlala** [comment]
Thanks for the clarification re: Stefan's statement on the mixed-patch subject. Let me give the implementation method some thought.

**Comment by andreasblueher** [comment]
I also left a comment under Stefans blog post so he might do some more clarification.

**Comment by brianlala** [comment]
For now, a workaround would be to grab both the July and the June update packages, into the same parent path (\Updates). Something like AutoSPUpdater would then first apply the lone July update, then attempt to install the language-independent June update (and gracefully exit because a newer patch is already install), then move on to the June language-dependent update (which would be installed). Then you'd have a "fully-patched", current farm.

**Comment by andreasblueher** [comment]
I contacted Stefan again on Twitter, since I never got a reply on his blog. The scenario where I ran into this was rather a new installation than an update scenario. This could would, but still how would source builder know when it has to pull the June package aswell?

Wouldn't it be possible to some code which would (using the version number) to get the latest language dependent and the lastest language independent package? Loading 2 language independent packages seems unnessary. If you like I could try to look into it, but it would require you to change the xml to have some indication on the file to seperate both? How do you like that idea?

**Comment by brianlala** [comment]
I found a simple fix for this. In the XML, in months were we don't have a language-dependent fix, I simply put a "fake" entry for the same month and build, but the download link and file name actually points to the last previously-released language-dependent updates. So for example, the SP2016 January update folder called "16.0.4795.1001 (January 2019)" actually contains the January 2019 language-independent (sts*.exe) update but as well has the October 2018 language-dependent (wss*.exe) update. 

I've updated the comments in the XML to allude to this as well. Nice part is no code changes were required!
